rfc/0009-audit-checks.md

```diff
@@ -35,67 +35,50 @@ As a Kubernetes developer, I want access to the audit checks results to allow me
 
```

Comment:
I think we should add a user story:

```suggestion
As a Kubernetes developer, in the future, I want to be able to add PolicyReportResults in real time for real requests and real policies.
```

I think we ought to make our future selves happier if we allow for Kubewarden to report all policy evaluation via PolicyReportResults in the future, not only audit ones.

Reply:
Do you mean also adding failed evaluation to the PolicyReportResults?

Reply:
I don't intend to add it now, but to have an implementation that is compatible with such. I would expect the same mechanism for normal requests, just that the PolicyReportResults would be in a different new PolicyReport and ClusterPolicyReport.

Even if we don't implement this in the future, it could use the same http rest API to report results. The results could be consumed by open-telemetry, or via a UI for example.

Reply:
Normal requests that were accepted will also be added to the PolicyReport, rejected requests for policies in monitor mode will also be added. Requests that were not accepted in enforce mode will not be added since these resources are not in the cluster. So why another PolicyReport and ClusterPolicyReport?

Reply:
Logging real requests being rejected into PolicyReports, or exposing that in an Rest API would be useful. Right now, one needs to go read the policy-server pod logs, and compile the tuple (PolicyServer, (Cluster)AdmissionPolicy, result), or deploy Jaeger.
Those results could be used in the UI for instantaneous feedback.

Reply:
I think this is something we should do in the future, but for now I would leave out of the scope of this RFC.

Reply:
Still, I'm just asking, if it's a good idea to provide an implementation that can support that use case in the future (like the user story).

Reply:
@flavio I would like to make the implementation compatible with adding real requests into `PolicyReport` and `ClusterPolicyReport` of their own.

Edited the suggestion for clarity.